From 4cd125baf55aba3a7483ca50aa99cbcbeaf617bb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Sun, 6 Sep 2020 01:39:42 +0200
Subject: [PATCH] Allow compilation with wxWidgets STL build

wxString may not be implicitly convertible to const char*, so add an
overload.
---
 src/limeRFE/limeRFE_wxgui.cpp | 7 +++++--
 src/limeRFE/limeRFE_wxgui.h   | 1 +
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/limeRFE/limeRFE_wxgui.cpp b/src/limeRFE/limeRFE_wxgui.cpp
index eec270dc..8a9cad7b 100644
--- a/src/limeRFE/limeRFE_wxgui.cpp
+++ b/src/limeRFE/limeRFE_wxgui.cpp
@@ -126,7 +126,10 @@ void limeRFE_wxgui::OnbtnClosePort(wxCommandEvent& event) {
 
 void limeRFE_wxgui::AddMssg(const char* mssg) {
 	wxString s(mssg, wxConvUTF8);
+	AddMssg(s);
+}
 
+void limeRFE_wxgui::AddMssg(const wxString& mssg) {
 	time_t rawtime;
 	struct tm * timeinfo;
 	char buffer[80];
@@ -134,7 +137,7 @@ void limeRFE_wxgui::AddMssg(const char* mssg) {
 	time(&rawtime);
 	timeinfo = localtime(&rawtime);
 	strftime(buffer, 80, "%H:%M:%S", timeinfo);
-	wxString line(wxString::Format("[%s] %s", buffer, s));
+	wxString line(wxString::Format("[%s] %s", buffer, mssg));
 
 	txtMessageField->AppendText(line + _("\n"));
 }
@@ -1114,4 +1117,4 @@ void limeRFE_wxgui::OncbEnableSWR(wxCommandEvent& event) {
 
 void limeRFE_wxgui::OnbtnClearMessages(wxCommandEvent& event) {
 	txtMessageField->Clear();
-}
\ No newline at end of file
+}
diff --git a/src/limeRFE/limeRFE_wxgui.h b/src/limeRFE/limeRFE_wxgui.h
index 95cd4176..623ad469 100644
--- a/src/limeRFE/limeRFE_wxgui.h
+++ b/src/limeRFE/limeRFE_wxgui.h
@@ -25,6 +25,7 @@ class limeRFE_wxgui : public limeRFE_view
 		void OnbtnOpenPort(wxCommandEvent& event);
 		void OnbtnClosePort(wxCommandEvent& event);
 		void AddMssg(const char* mssg);
+		void AddMssg(const wxString& mssg);
 		void ReadPorts();
 		void OnbtnRefreshPorts(wxCommandEvent& event);
 		void OnbtnReset(wxCommandEvent& event);
-- 
2.28.0

